source/common/upstream/eds.cc

```diff
@@ -69,22 +69,27 @@ void EdsClusterImpl::onConfigUpdate(const ResourceVector& resources) {
       throw EnvoyException(
           fmt::format("Unexpected non-zero priority for local cluster '{}'.", cluster_name_));
     }
-    if (new_hosts.size() <= priority) {
-      new_hosts.resize(priority + 1);
+    if (priority_state.size() <= priority) {
+      priority_state.resize(priority + 1);
     }
-    if (new_hosts[priority] == nullptr) {
-      new_hosts[priority] = HostListPtr{new HostVector};
+    if (priority_state[priority].first == nullptr) {
+      priority_state[priority] = {std::make_unique<HostVector>(), {}};
+    }
+    if (locality_lb_endpoint.has_locality() && locality_lb_endpoint.has_load_balancing_weight()) {
+      priority_state[priority].second[locality_lb_endpoint.locality()] =
```

Comment:
This one I'll leave as is, since it's local to this function, so it would be probably less readable to start defining new types. I've been generally trying to avoid the contagion of vectors at the interface level.